src/fsharp/vs/ServiceNavigation.fs

```diff
@@ -191,23 +193,42 @@ module NavigationImpl =
-                | SynMemberDefn.Member(bind, _) -> processBinding true bind
-                | SynMemberDefn.ValField(Field(_, _, Some(rcid), ty, _, _, _, _), _) ->
-                    [ createMember(rcid, FieldDecl, GlyphMajor.FieldBlue, ty.Range) ]
-                | SynMemberDefn.AutoProperty(_attribs,_isStatic,id,_tyOpt,_propKind,_,_xmlDoc,_access,_synExpr, _, _) -> 
-                    [ createMember(id, FieldDecl, GlyphMajor.FieldBlue, id.idRange) ]
-                | SynMemberDefn.AbstractSlot(ValSpfn(_, id, _, ty, _, _, _, _, _, _, _), _, _) ->
-                    [ createMember(id, MethodDecl, GlyphMajor.Method2, ty.Range) ]
-                | SynMemberDefn.NestedType _ -> failwith "tycon as member????" //processTycon tycon                
-                | SynMemberDefn.Interface(_, Some(membs), _) ->
-                    processMembers membs |> snd
-                | _ -> []  )) 
-            ((members |> Seq.map fst |> Seq.fold unionRangesChecked range.Zero),
-             (members |> List.map snd |> List.concat))
+            let members = 
+                members 
+                |> List.groupBy (fun x -> x.Range)
+                |> List.map (fun (range, members) ->
+                    range,
+                    (match members with
+                     | [memb] ->
+                         match memb with
+                         | SynMemberDefn.LetBindings(binds, _, _, _) -> List.collect (processBinding false) binds
+                         | SynMemberDefn.Member(bind, _) -> processBinding true bind
+                         | SynMemberDefn.ValField(Field(_, _, Some(rcid), ty, _, _, access, _), _) ->
+                             [ createMember(rcid, FieldDecl, GlyphMajor.FieldBlue, ty.Range, access) ]
+                         | SynMemberDefn.AutoProperty(_attribs,_isStatic,id,_tyOpt,_propKind,_,_xmlDoc, access,_synExpr, _, _) -> 
+                             [ createMember(id, FieldDecl, GlyphMajor.FieldBlue, id.idRange, access) ]
+                         | SynMemberDefn.AbstractSlot(ValSpfn(_, id, _, ty, _, _, _, _, access, _, _), _, _) ->
+                             [ createMember(id, MethodDecl, GlyphMajor.Method2, ty.Range, access) ]
+                         | SynMemberDefn.NestedType _ -> failwith "tycon as member????" //processTycon tycon                
+                         | SynMemberDefn.Interface(_, Some(membs), _) ->
+                             processMembers membs |> snd
+                         | _ -> [] 
+                     // can happen if one is a getter and one is a setter
+                     | [SynMemberDefn.Member(memberDefn=Binding(headPat=SynPat.LongIdent(lid1, Some(info1),_,_,_,_)) as binding1)
+                        SynMemberDefn.Member(memberDefn=Binding(headPat=SynPat.LongIdent(lid2, Some(info2),_,_,_,_)) as binding2)] ->
+                         // ensure same long id
+                         assert((lid1.Lid,lid2.Lid) ||> List.forall2 (fun x y -> x.idText = y.idText))
```

Comment:
I've ported this code from here https://github.com/Microsoft/visualfsharp/blob/master/src/fsharp/vs/ServiceParseTreeWalk.fs#L418-L454 I think that if this assertion violated, it means that something very wrong happened with the parser and we must see it in debug configuration.